drivers/gpu/drm/lima/lima_drv.c

```diff
@@ -200,7 +202,20 @@ static const struct file_operations lima_drm_driver_fops = {
 };
 
 static struct drm_driver lima_drm_driver = {
-	.driver_features    = DRIVER_RENDER | DRIVER_GEM,
+	.driver_features    = DRIVER_RENDER | DRIVER_GEM | DRIVER_PRIME,
+
+	/* PRIME Operations */
+	.gem_prime_get_sg_table    = drm_gem_cma_prime_get_sg_table,
+	.gem_prime_import_sg_table = drm_gem_cma_prime_import_sg_table,
+	.gem_prime_vmap            = drm_gem_cma_prime_vmap,
+	.gem_prime_vunmap          = drm_gem_cma_prime_vunmap,
+	.gem_prime_mmap            = drm_gem_cma_prime_mmap,
+
+	.prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+	.prime_fd_to_handle = drm_gem_prime_fd_to_handle,
+	.gem_prime_export   = drm_gem_prime_export,
```

Comment:
Have you tested the export function? I believe it'll just crash if user call it. If not implement correctly, you can just not assign this function pointer so that user won't call into it.

Reply:
Also if drop this function, gem_prime_get_sg_table is not needed too.

Reply:
So does the gem_prime_vmap/gem_prime_vunmap/gem_prime_mmap/prime_handle_to_fd